lib/spack/spack/database.py

```diff
@@ -1653,5 +1654,5 @@ class InvalidDatabaseVersionError(SpackError):
     def __init__(self, expected, found):
         super(InvalidDatabaseVersionError, self).__init__(
             "Expected database version %s but found version %s." % (expected, found),
-            "`spack reindex` may fix this, or you may need a newer " "Spack version.",
+            "\n\t`spack reindex` may fix this, or you may need a newer Spack version.\n",
```

Comment:
I think this should just say:

```
You need a newer Spack version to read this metadata.
```

Ideally it would also say what you're trying to read in the error message (i.e., buildcache or DB) -- I don't think it does currently.

`spack reindex` is really only something devs will know about or use -- I think that's only going to come up if you remove the new installation and run `spack reindex`.  We should probably have a page like this one, but for databases:

* https://spack.readthedocs.io/en/latest/spack.environment.html#spack-lock-format

And just say "see <URL> for details", then provide details like the `spack reindex` thing there.